PennyPincher.cs

```diff
@@ -244,9 +245,14 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
                 if (i == listing.ItemListings.Count) return;
             }
 
-            var price = listing.ItemListings[i].PricePerUnit - (listing.ItemListings[i].PricePerUnit % configuration.mod) - configuration.delta;
-            price -= (price % configuration.multiple);
-            price = Math.Max(price, configuration.min);
+            long price = listing.ItemListings[i].PricePerUnit;
+            if (!IsOwnRetainer(listing.ItemListings[i].RetainerId))
+            {
+                price = price - (listing.ItemListings[i].PricePerUnit % configuration.mod) - configuration.delta;
+                price -= (price % configuration.multiple);
+                price = Math.Max(price, configuration.min);
+            }
+
```

Comment:
This is confusing to me - it doesn't seem like you ever look at ItemListings[i+1]. Can you use something like

```
while (i < listing.ItemListings.Count && IsOwnRetainer(listing.ItemListings[i].RetainerId)) i++;
if (i == listing.ItemListings.Count) return;
```
instead?

Reply:
Updated.